evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -173,7 +174,11 @@ def transform(self, X, y=None):
         if self._numeric_cols is not None and len(self._numeric_cols) > 0:
             X_numeric = X.ww[self._numeric_cols.tolist()]
             imputed = self._numeric_imputer.transform(X_numeric)
-            X_no_all_null[X_numeric.columns] = imputed
+            for numeric_col in X_numeric.columns:
+                X_no_all_null.ww[numeric_col] = init_series(
+                    imputed[numeric_col],
+                    logical_type="Double",
+                )
```

Comment:
Initializing each of these imputed columns as a series with a `Double` logical type (because they belong to `X_numeric.columns`, led to another large `fit` time improvement